tools/run_tests/run_xds_tests.py

```diff
@@ -1586,6 +1588,181 @@ def test_timeout(gcp, original_backend_service, instance_group):
         patch_url_map_backend_service(gcp, original_backend_service)
 
 
+def test_fault_injection(gcp, original_backend_service, instance_group):
+    logger.info('Running test_fault_injection')
+
+    logger.info('waiting for original backends to become healthy')
+    wait_for_healthy_backends(gcp, original_backend_service, instance_group)
+
+    testcase_header = 'fi_testcase'
+
+    def _route(pri, name, fi_policy):
+        return {
+            'priority': pri,
```

Comment:
This is great that we only update the urlMap once. But I don't understand why this work? How do we get the client pick the corresponding rule?

Reply:
The header matcher selects the appropriate one.  From the comment:

```py
    # ...  Each
    # test case will set the testcase_header with the testcase_name for routing
    # to the appropriate config for the case, defined above.
```

Reply:
Got it.